src/StartGamePacket.php

```diff
@@ -150,29 +163,66 @@ protected function decodePayload(PacketSerializer $in) : void{
 		$this->worldName = $in->getString();
 		$this->premiumWorldTemplateId = $in->getString();
 		$this->isTrial = $in->getBool();
-		$this->playerMovementSettings = PlayerMovementSettings::read($in);
+		if($in->getProtocolId() >= ProtocolInfo::PROTOCOL_1_13_0){
+			$this->playerMovementSettings = PlayerMovementSettings::read($in);
+		}
 		$this->currentTick = $in->getLLong();
 
 		$this->enchantmentSeed = $in->getVarInt();
 
 		$this->blockPalette = [];
-		for($i = 0, $len = $in->getUnsignedVarInt(); $i < $len; ++$i){
-			$blockName = $in->getString();
-			$state = $in->getNbtCompoundRoot();
-			$this->blockPalette[] = new BlockPaletteEntry($blockName, new CacheableNbt($state));
+		if($in->getProtocolId() >= ProtocolInfo::PROTOCOL_1_13_0){
+			if($in->getProtocolId() >= ProtocolInfo::PROTOCOL_1_16_100){
+				for($i = 0, $len = $in->getUnsignedVarInt(); $i < $len; ++$i){
+					$blockName = $in->getString();
```

Comment:
would be nice if this was extracted to a different function, since it'll start to look like a mess otherwise